src/Land/Model/SoilHeatParameterizations.jl

```diff
@@ -0,0 +1,196 @@
+    T = T_ref + (I + θ_i * ρ_i * LH_f0) / cs
+    return T
+end
+
+"""
+    volumetric_heat_capacity(
+        θ_l::FT,
+        θ_i::FT,
+        c_ds::FT,
+        cp_l::FT,
+        cp_i::FT
+    ) where {FT}
+
+Compute the expression for volumetric heat capacity.
+"""
+function volumetric_heat_capacity(
+    θ_l::FT,
+    θ_i::FT,
+    c_ds::FT,
+    cp_l::FT,
+    cp_i::FT,
+) where {FT}
+    c_s = c_ds + θ_l * cp_l + θ_i * cp_i
+    return c_s
+end
+
+"""
+    internal_energy(
+        θ_i::FT,
+        c_s::FT,
+        T::FT,
+        T_ref::FT,
+        ρ_i::FT,
+        LH_f_0::FT
+    ) where {FT}
+Compute the expression for internal energy.
+"""
+function internal_energy(
+    θ_i::FT,
+    c_s::FT,
+    T::FT,
+    T_ref::FT,
+    ρ_i::FT,
+    LH_f_0::FT,
+) where {FT}
+    I = c_s * (T - T_ref) - θ_i * ρ_i * LH_f_0
+    return I
+end
+
+"""
+    saturated_thermal_conductivity(
+        θ_l::FT,
+        θ_i::FT,
+        κ_sat_unfrozen::FT,
+        κ_sat_frozen::FT
+    ) where {FT}
+Compute the expression for saturated thermal conductivity of soil matrix.
+"""
+function saturated_thermal_conductivity(
+    θ_l::FT,
+    θ_i::FT,
+    κ_sat_unfrozen::FT,
+    κ_sat_frozen::FT,
+) where {FT}
+    #TBD: can we get rid of this branch? if not: create test for it.
+    θ_w = θ_l + θ_i
+    if θ_w < eps(FT)
+        κ_sat = FT(0.0)
+    else
+        κ_sat = FT(κ_sat_unfrozen^(θ_l / θ_w) * κ_sat_frozen^(θ_i / θ_w))
+    end
+
+    return κ_sat
+end
+
+"""
+    relative_saturation(
+            θ_l::FT,
+            θ_i::FT,
+            porosity::FT
+    ) where {FT}
+Compute the expression for relative saturation.
+"""
+function relative_saturation(θ_l::FT, θ_i::FT, porosity::FT) where {FT}
+    S_r = (θ_l + θ_i) / porosity
+    return S_r
+end
+"""
+    kersten_number(
+        θ_i::FT,
+        S_r::FT,
+        a::FT,
+        b::FT,
+        ν_om::FT,
+        ν_sand::FT,
+        ν_gravel::FT
+    ) where {FT}
+Compute the expression for the Kersten number.
+"""
+function kersten_number(
+    θ_i::FT,
+    S_r::FT,
+    a::FT,
+    b::FT,
+    ν_om::FT,
+    ν_sand::FT,
+    ν_gravel::FT,
+) where {FT}
+
+    if θ_i < eps(FT)
+        K_e =
+            S_r^((FT(1) + ν_om - a * ν_sand - ν_gravel) / FT(2)) *
+            (
+                (FT(1) + exp(-b * S_r))^(-FT(3)) -
+                ((FT(1) - S_r) / FT(2))^FT(3)
+            )^(FT(1) - ν_om)
+    else
+        K_e = S_r^(FT(1) + ν_om)
+    end
+    return K_e
+end
+
+"""
+    thermal_conductivity(
+        κ_dry::FT,
+        K_e::FT,
+        κ_sat::FT
+    ) where {FT}
+Compute the expression for thermal conductivity of soil matrix.
+"""
+function thermal_conductivity(κ_dry::FT, K_e::FT, κ_sat::FT) where {FT}
+    κ = K_e * κ_sat + (FT(1) - K_e) * κ_dry
+    return κ
+end
+
+"""
+    internal_energy_liquid_water(
+        cp_l::FT,
+        T::FT,
+        T_ref::FT,
+        ρ_l::FT
+    ) where {FT}
+Compute the expression for the internal energy of liquid water.
```

Comment:
this is picky, but can we label this as "volumetric_internal_energy"?

Reply:
(in the comments is fine)

Reply:
Also, we might note that cp_l is the volumetric heat capacity of liquid water

Reply:
done